src/test/java/org/jabref/logic/shared/SynchronizationTestSimulator.java

```diff
@@ -32,23 +32,27 @@
 @DatabaseTest
 public class SynchronizationTestSimulator {
 
-    @Parameter
     public DBMSType dbmsType;
     private BibDatabaseContext clientContextA;
     private BibDatabaseContext clientContextB;
     private SynchronizationTestEventListener eventListenerB; // used to monitor occurring events
-    private DBMSConnection dbmsConnection;
-
-    @Parameters(name = "Test with {0} database system")
-    public static Collection<DBMSType> getTestingDatabaseSystems() {
-        return TestManager.getDBMSTypeTestParameter();
+    private final GlobalBibtexKeyPattern pattern = GlobalBibtexKeyPattern.fromPattern("[auth][year]");
+
+    private static Stream<Object[]> getTestingDatabaseSystems() throws InvalidDBMSConnectionPropertiesException, SQLException {
+        Collection<Object[]> result = new ArrayList<>();
+        for (DBMSType dbmsType : TestManager.getDBMSTypeTestParameter()) {
+            result.add(new Object[] {
+                                     dbmsType,
+                                     TestConnector.getTestDBMSConnection(dbmsType),
+                                     DBMSProcessor.getProcessorInstance(TestConnector.getTestDBMSConnection(dbmsType))});
+        }
+        return result.stream();
     }
 
     @BeforeEach
-    public void setUp() throws SQLException, DatabaseNotSupportedException, InvalidDBMSConnectionPropertiesException {
-        this.dbmsConnection = TestConnector.getTestDBMSConnection(dbmsType);
+    @MethodSource("getTestingDatabaseSystems")
+    public void setUp(DBMSType dbmsType, DBMSConnection dbmsConnection, DBMSProcessor dbmsProcessor) throws SQLException, DatabaseNotSupportedException, InvalidDBMSConnectionPropertiesException {
```

Comment:
Does this really works, i.e. did you tested that all test methods are invoked correctly? According to the documentation the `before/aftereach` handlers don't get their parameterized resolved.
> Since a test class may contain regular tests as well as parameterized tests with different parameter lists, values from argument sources are not resolved for lifecycle methods (e.g. @BeforeEach) and test class constructors.

But this is in a slightly different context. If this method works, then it should also be used in `DBMSSynchronizerTest`.

Reply:
Well I will see if I can test it locally, but otherwise Travis is green and so I thought it's working

Reply:
Without double chcking (but trying to understand the implications of Tobias' explanation), I would bet that the test use only one database system and not all avaiable.